<launch>

  <arg name="gui" default="false" />
  <arg name="js_pub_rate" default="100" />
  <arg name="circle_period" default="10" />
  <arg name="tf_pub_rate" default="15" />
  <arg name="pos_pub_rate" default="100" />
  <arg name="rvizconfig" default="$(find bwrobot)/rviz/rvizconfig.rviz" />

  <param name="robot_description" textfile="$(find bwrobot)/urdf/2link_robot.urdf" />
  <param name="use_gui" value="$(arg gui)"/>

  <node name="controller_node" pkg="bwrobot" type="SOLGPR_ROS.py" unless="$(arg gui)">
  	<param name="controller_pub_rate" value="$(arg js_pub_rate)" />
  	<param name="period" value="$(arg circle_period)" />
  </node>

  <node if="$(arg gui)" name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" />
  
  <node name="Pose_node" pkg="bwrobot" type="PoseTalk.py">
  	<param name="tf_ee_pub_rate" value="$(arg pos_pub_rate)" />
  </node>

  <node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher" />

  <node name="rviz" pkg="rviz" type="rviz" args="-d $(arg rvizconfig)" required="true" />

</launch>
